genie-core/src/main/java/com/netflix/genie/core/jpa/entities/ApplicationEntity.java

```diff
@@ -113,60 +120,66 @@ protected void onCreateOrUpdateApplication() throws GenieException {
     }
 
     /**
-     * Sets the configurations for this application.
+     * Set all the files associated as configuration files for this application.
      *
-     * @param configs The configuration files that this application needs
+     * @param configs The configuration files to set
      */
-    public void setConfigs(final Set<String> configs) {
+    public void setConfigs(@Nullable final Set<FileEntity> configs) {
```

Comment:
If updating an attached application entity, then the logic of clear and adding will end up deleting all records and inserting the new ones. Should be fine?

Reply:
I'm not clear on how it works. My assumption was that since this modification will be called inside the scope of a transaction at the end of the transaction only the final state will be translated into the SQL calls not the intermediate changes. I don't think it acts like a write ahead log. I could be wrong but this seemed the easiest way to do a set differential. Open to other ideas?